src/server.ts

```diff
@@ -34,15 +34,10 @@ const clientLib = path.resolve(__dirname, '..', 'client', 'lib');
 export class Server {
   readonly url: string;
   private readonly server: net.Server;
-
-  // Even though we're running benchmarks in series, we give each run an id and
-  // make sure that we associate result messages with the correct run. This
-  // prevents any spurrious race conditions and enables one runner to launch
-  // multiple clients eventually.
-  private currentRunId = 0;
-  private currentRunBytes = 0;
-  private readonly pendingRuns = new Map<string, PendingBenchmark>();
-  private resultSubmitted = new Deferred<BenchmarkResult>();
+  private sessionPending = false;
+  private sessionBytes = 0;
+  private sessionUserAgent = '';
```

Comment:
Consider making a Session class to encapsulate this per-session data, just for clarity.

Reply:
Done.